### DISCUSSION TITLE
dateFormat not being applied correctly on Calendar Component

### DISCUSSION THREAD

**Discussion by Fluxstone** [primefaces/community#460] [PrimeVue] [answered]
**Problem**

I am using the Calendar Component which has a v-model to a variable `hotelArrivalDate`. This is a string in the format `yyyy-mm-dd`. However I don't want it displayed as such and instead want the component to display it like `dd.mm.yyyy` -> Therefor I use the dateFormat property like `dateFormat="dd.mm.yy"`. This sadly does not change the format however.

![image](https://github.com/primefaces/community/assets/41082720/ced791c4-040b-4c36-b93a-3f0397df85a2)
_Top Value is what the text should look like. It takes the v-model variable `hotelArrivalDate ` and applies string manipulation. Bottom Value is the v-model variable which is also being displayed inside the component._

**Code**

```
<Calendar 
  dateFormat="dd.mm.yy"
  v-model="hotelArrivalDate"
  class="w-full"
  :placeholder='translation.views.hotel.hotel_arrival_date'
  showIcon
  @date-select="changeFieldValue('hotelArrivalDate', hotelArrivalDate)"
/>

<!-- Displaying the debug values. Top with string manipulation, bottom the v-model variable -->
{{(hotelArrivalDate).split('-').map(Number).reverse().join('.')}} <br>
{{hotelArrivalDate}} <br>
```
![2023-11-14 11 44 33](https://github.com/primefaces/community/assets/41082720/491d68de-56ed-48d3-8ed1-b40eb6f497e5)
_Interestingly, when changing the date the correct format can be seen for a split second before reverting back to the v-model format._

**Question**

How can I display the date in my desired format without messing with the v-model variable? It is used across a few things so it cannot be changed.
One way I thought of would be to maybe write some sort of function and supply an additional variables that is used instead of `hotelArrivalDate`. Whenever the date is updated the function would do the string manipulation, save it once in `dd.mm.yyyy` and in `yyyy-mm-dd` for my system to use. However there has to be an easier way than that no?

Kind regards and thanks for any help for this Vue beginner :)

**Comment by tugcekucukoglu** [comment]
It seems this implementation is correct: ` dateFormat="dd.mm.yy"`. How can I replicate the issue? Could you send a reproducer demo link?

**Comment by Fluxstone** [comment, marked as answer]
Sorry for the late reply.

Thank you tugcekucukoglu for taking the time to look over this.

After creating the stackblitz (which worked as expected unlike my example) someone from our team looked over it and we found the issue:

In this case, whenever information was entered, there was a page reload. On page reload the data (which came from a parent component) was already handed over formatted in `yyyy-mm-dd`. So the `dateFormat` feature worked as expected, the content of the inputText was just overridden (as could be seen in the gif, the format is shortly in the correct format, then the page reload happens and its parsed as yyyy-mm-dd).

A oversight on my end. I apologise. Thank you for getting me on the right track however!
